[Review] : 리뷰 조회 기능 구현

rlqor9688 opened the pull request:
## Description
리뷰 조회 기능을 구현하였습니다.

<요구사항>
1. 리뷰는 단건 조회할 수 없습니다.
2. 리뷰는 가게 정보를 기준으로 다건 조회 가능하며, 최신순으로 정렬합니다.
3. 리뷰를 별점 범위에 따라 조회할 수 있습니다.- ex) 3~5점

## Changes
1. ReviewController : findAllReviewByStore 메서드 추가
2. ReviewRepository : List<Reviews> findAllByStores_IdAndRatingBetween(Long storesId, int minRating, int maxRating) 조회 로직 추가
3. findAllReviewByStore 메서드 추가
4. ReviewServiceFindTest 클래스 내 테스트코드 구현(Store 미존재 시 / Store 폐업 상태 시 예외처리)
5. ReviewServiceUpdateTest 내 미사용 import 삭제

## Ref
![image](https://github.com/user-attachments/assets/8255462a-f049-4908-8972-05d91fd8e90d)

rlqor9688 committed:
feat [review] : 리뷰 조회 기능 구현

1. ReviewController : findAllReviewByStore 메서드 추가

2. ReviewRepository : List<Reviews> findAllByStores_IdAndRatingBetween(Long storesId, int minRating, int maxRating) 조회 로직 추가;

3. findAllReviewByStore 메서드 추가

4. ReviewServiceFindTest 클래스 내 테스트코드 구현

5. ReviewServiceUpdateTest 내 미사용 import 삭제

rlqor9688 committed:
fix [review] : 리뷰 조회 기능 내림차순 정렬 조건 추가

1. ReviewRepository : 내림차순 JPA 쿼리 조건 추가(조회 메서드명 수정)

2. ReviewServiceFindTest : repository 메서드명 수정

3. ReviewServiceImpl : repository 메서드명 수정

rlqor9688 committed:
Merge branch 'dev' into feature/kiback-review

rlqor9688 committed:
Merge branch 'dev' into feature/kiback-review

nagul2 commented on src/main/java/com/delivery/igo/igo_delivery/api/review/controller/ReviewController.java:
```diff
@@ -38,9 +38,12 @@ public ResponseEntity<Void> updateReview(
-//    public ResponseEntity<List<ReviewResponseDto>> findAllReviewByStore(@RequestParam Long storesId) {
-//        return new ResponseEntity<>(reviewService.findAllReviewByStore(storesId), HttpStatus.OK);
-//    }
+    @GetMapping
+    public ResponseEntity<List<ReviewResponseDto>> findAllReviewByStore(
+            @RequestParam Long storesId,
+            @RequestParam(defaultValue = "1") int minRating,
+            @RequestParam(defaultValue = "5") int maxRating) {
+        return new ResponseEntity<>(reviewService.findAllReviewByStore(storesId, minRating, maxRating), HttpStatus.OK);
+    }
```
일단 구현하셨으니 넘어가셔도 되긴 하는데, 애플리케이션에서 별점 많은순 낮은 순을 정렬을 선택해서 리뷰를 보는 경우는 본 것 같은데, 사용자가 일부의 3 ~ 5의 별점만 출력되는 기능이 필요한지...는 잘 모르겠습니다

혹시 사례같은게 있다면 공유 부탁드릴게요

rlqor9688 replied:
아하~ 요구사항에 아래와 같은 조건이 있어 위와 같이 구현했습니다!
- 리뷰를 별점 범위에 따라 조회할 수 있습니다.
    - ex) 3~5점

rlqor9688 replied:
아마 실제 서비스에서는 별점 몇 점 이상인 것 보기 기능이지 않을까 싶네요. minRating이 사용자가 선택하는 값이고 maxRating은 default로 5점이 들어가도록 되어 있는 방식이 아닐까 싶습니다

nagul2 commented on src/main/java/com/delivery/igo/igo_delivery/api/review/service/ReviewServiceImpl.java:
```diff
@@ -116,14 +117,22 @@ public void updateReview(Long reviewId, AuthUser authUser, ReviewUpdateRequestDt
+            throw new GlobalException(ErrorCode.REVIEW_STORE_IS_CLOSED);
+        }
+
+        List<Reviews> reviewList = reviewRepository.findAllByStores_IdAndRatingBetweenOrderByCreatedAtDesc(storesId, minRating, maxRating);
+
```
와우,,, 
제가알기론 이정도 길어지면 직접 쿼리를 작성하는게 가독성이 좋다는 점만 참고해주시면 좋을 것 같습니다.
시간이 없으니 패스!!

rlqor9688 replied:
감사합니다! 안 그래도 고민했는데, 수정했습니다!

rlqor9688 committed:
Merge branch 'dev' into feature/kiback-review

rlqor9688 committed:
docs [review] : repository 메서드 수정 / 최신 dev pull 후 re-commit&push

1. ReviewRepository 메서드명 수정(쿼리 사용)

2. ReviewServiceImpl 메서드명 수정

3. ReviewServiceFindTest 메서드명 수정
